test/unit/core/network/createNetwork.spec.js

```diff
@@ -161,62 +165,44 @@ describe("createNetwork", () => {
       onBeforeSend: () => undefined,
       onResponse: response => {
-        const cleanResponse = JSON.parse(JSON.stringify(response));
+        const cleanResponse = response.toJSON();
         expect(cleanResponse).toEqual(myresponse);
         done();
       }
     };
-    const networkStrategy = () => {
-      return new Promise(resolve => {
-        resolve(JSON.stringify(myresponse));
-      });
-    };
-    const network = createNetwork(config, logger, lifecycle, networkStrategy);
-    network.newRequest().send();
+    const networkStrategy = () => Promise.resolve(JSON.stringify(myresponse));
+    createNetwork(config, logger, lifecycle, networkStrategy)
+      .newRequest()
+      .send();
   });
 
-  [true, false].forEach(b => {
-    it(`allows components to get the request info (beacon = ${b})`, done => {
-      let requestPayload;
-      let requestPromise;
+  [true, false].forEach(expectsResponse => {
+    it(`allows components to get the request info (beacon = ${expectsResponse})`, done => {
+      let lifecyclePayload;
+      let lifecycleResponsePromise;
+      let lifecycleIsBeacon;
       const lifecycle = {
-        onBeforeSend: ({ payload, responsePromise, isBeacon, send }) => {
-          expect(send).toBeUndefined();
-          expect(payload).toBe(requestPayload);
-          expect(responsePromise).toBe(requestPromise);
-          expect(isBeacon).toBe(b);
+        onBeforeSend: ({ payload, responsePromise, isBeacon }) => {
+          lifecyclePayload = payload;
+          lifecycleResponsePromise = responsePromise;
+          lifecycleIsBeacon = isBeacon;
           done();
-        },
-        onResponseFragment: () => undefined
+        }
       };
       const networkStrategy = () => new Promise(() => undefined);
       const network = createNetwork(config, logger, lifecycle, networkStrategy);
-      const { payload, responsePromise, send } = network.newRequest(b);
-      requestPayload = payload;
-      requestPromise = responsePromise;
-      send();
+      const { payload, send } = network.newRequest(expectsResponse);
+      const responsePromise = send();
+      responsePromise.then(() => {
+        expect(lifecyclePayload).toBe(payload);
+        expect(lifecycleResponsePromise).toBe(responsePromise);
+        expect(lifecycleIsBeacon).toBe(expectsResponse);
+        expect(responsePromise).toBe(lifecycleResponsePromise);
+        expect(expectsResponse).toBe(expectsResponse);
+      });
     });
   });
 
-  it("logs json parse errors", done => {
-    const networkStrategy = () => {
-      return Promise.resolve("badbody1");
-    };
-    const loggerSpy = jasmine.createSpyObj("logger", ["warn"]);
-    const network = createNetwork(
-      config,
-      loggerSpy,
-      nullLifecycle,
-      networkStrategy
-    );
-    const { send } = network.newRequest();
-    send();
-    setTimeout(() => {
-      expect(loggerSpy.warn).toHaveBeenCalledTimes(1);
-      done();
-    }, 100);
-  });
-
```

Comment:
We were logging parse errors but instead it made sense to throw an error. That error will bubble up, get augmented with additional context along the way, and then the user can either handle the error in a `catch` or, if they don't, it will be logged to the console by virtue of it being an unhandled promise exception.